lib/ansible/modules/crypto/openssl_cert.py

```diff
@@ -0,0 +1,446 @@
+        description:
+            - Remote absolute path where the generated certificate file should be created.
+              If the parent directories do not exist, they are automatically created as long as the user has sufficient privileges.
+    provider:
+        required: true
+        choices: [ 'selfsigned', 'assertonly', 'letsencrypt' ]
+        description:
+            - Name of the provider to use to generate/retrieve the OpenSSL certificate.
+              The 'assertonly' provider will not generate files and fail if the certificate file is missing.
+    force:
+        required: false
+        default: False
+        choices: [ True, False ]
+        description:
+            - Generate the certificate, even if it already exists.
+    backup:
+        required: false
+        default: False
+        choices: [ True, False ]
+        description:
+            - Create a backup file including timestamp information in case you were overwriting an existing certificate.
+    csr:
+        required: false
+        description:
+            - Path to the Certificate Signing Request (CSR) used to generate this certificate. This is not required in 'assertonly' mode.
+
+    signature_algorithms:
+        required: false
+        description:
+            - list of algorithms that you would accept the certificate to be signed with
+              (e.g. ['sha256WithRSAEncryption', 'sha512WithRSAEncryption'])
+    privatekey:
+        required: false
+        description:
+            - Path to the private key to use when signing the certificate
+    issuer:
+        required: false
+        description:
+            - Key/value pairs that must be present in the issuer name field of the certificate
+    subject:
+        required: false
+        description:
+            - Key/value pairs that must be present in the subject name field of the certificate
+    version:
+        required: false
+        description:
+            - Version of the certificate
+
+    selfsigned_digest:
+        required: false
+        default: "sha256"
+        description:
+            - Digest algorithm to be used when self-signing the certificate
+    selfsigned_notBefore:
+        required: false
+        default: 0
+        description:
+            - Number of seconds from now which the self-signed certificate will be valid from.
+    selfsigned_notAfter:
+        required: false
+        default: 315360000 (3650 days)
+        description:
+            - Number of seconds from now which the self-signed certificate will be expired.
+
+    letsencrypt_accountkey:
+        required: false
+        description:
+            - Path to the accountkey for the 'letsencrypt' provider
+    letsencrypt_challenge_path:
+        required: false
+        description:
+            - Path to the ACME challenge directory that is served on http://<HOST>:80/.well-known/acme-challenge/
+'''
+
+
+EXAMPLES = '''
+# Generate a Self Signed OpenSSL certificate
```

Comment:
Please use `-name: ` rather than a comment, as per best practice for writing playbooks

http://docs.ansible.com/ansible/dev_guide/developing_modules_documenting.html#examples-block